File: app/src/main/java/pl/softfly/flashcards/entity/deck/DeckConfig.java
package pl.softfly.flashcards.entity.deck;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * @author Grzegorz Ziemski
 */
@Entity(tableName = "Core_DeckConfig")
public class DeckConfig {

    public static final String FILE_SYNC_EDITING_BLOCKED_AT = "FileSync_EditingBlockedAt";

    public static final String STUDY_CARD_TERM_FONT_SIZE = "StudyCard_Term_FontSize";

    public static final String STUDY_CARD_DEFINITION_FONT_SIZE = "StudyCard_Definition_FontSize";

    public final static int STUDY_CARD_FONT_SIZE_DEFAULT = 24;

    public static final String STUDY_CARD_TD_DISPLAY_RATIO = "StudyCard_TD_DisplayRatio";

    public static final String MAX_FORGOTTEN_CARDS = "StudyCard_MaxForgottenCards";

    public static final int MAX_FORGOTTEN_CARDS_DEFAULT = 20;

    @NonNull
    @PrimaryKey
    private String key;

    private String value;

    public DeckConfig() {
    }

    public DeckConfig(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: room/src/main/java/pl/gocards/room/entity/deck/DeckConfig.kt
package pl.gocards.room.entity.deck

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author Grzegorz Ziemski
 */
@Entity(tableName = "Core_DeckConfig")
@SuppressWarnings("unused")
data class DeckConfig(
    @PrimaryKey
    var key: String,
    var value: String
) {
    companion object {
        const val FILE_SYNC_EDITING_BLOCKED_AT = "FileSync_EditingBlockedAt"
        const val STUDY_CARD_TERM_FONT_SIZE = "StudyCard_Term_FontSize"
        const val STUDY_CARD_DEFINITION_FONT_SIZE = "StudyCard_Definition_FontSize"
        const val STUDY_CARD_FONT_SIZE_DEFAULT = 24
        const val STUDY_CARD_FONT_SIZE_MIN = 10
        const val STUDY_CARD_FONT_SIZE_MAX = 50
        const val MAX_ALLOWED_FORGOTTEN_CARDS = "StudyCard_MaxForgottenCards"
        const val MAX_ALLOWED_FORGOTTEN_CARDS_DEFAULT = 5
        const val MAX_LINES = "ListCards_MaxLines"
        const val MAX_LINES_DEFAULT = 4
        const val AI_LAST_UPDATED_AT = "AI_LastUpdatedAt"
    }
}